x-pack/platform/plugins/shared/cloud/common/types.ts

```diff
@@ -8,3 +8,20 @@
 export interface ElasticsearchConfigType {
   elasticsearch_url?: string;
 }
+
+export type SolutionType = 'search' | 'elasticsearch' | 'observability' | 'security';
+export interface CloudDataAttributes {
+  onboardingData: {
+    solutionType?: SolutionType;
+    token: string;
+    security?: CloudSecurityAnswer;
+  };
+}
+
+export interface CloudSecurityAnswer {
+  useCase: 'siem' | 'cloud' | 'edr' | 'other';
+  migration?: {
+    value: boolean;
+    type?: 'splunk' | 'other';
+  };
+}
```

Comment:
Types moved from [x-pack/platform/plugins/shared/cloud/server/routes/types.ts](https://github.com/elastic/kibana/pull/210921/files#diff-c01bc9f98f018b181a5d5d62d9e5f1f8eaa9b0f9e1edf18aad17be0139876900) to a _/common_ directory, since we need to use this type in the UI of SecuritySolution plugin